docs/src/references/mappers/gen-mapper.md

```diff
@@ -0,0 +1,168 @@
+
+```toml
+input_topics = ["te/+/+/+/+/m/+"]
+
+stages = [
+    { filter = "add_timestamp.js" },
+    { filter = "drop_stragglers.js", config = { max_delay = 60 } },
+    { filter = "te_to_c8y.js", meta_topics = ["te/+/+/+/+/m/+/meta"] }
+]
+```
+
+- A filter has to export at least one `process` function.
+  - `process(t: Timestamp, msg: Message, config: Json) -> Vec<Message>` 
+  - This function is called for each message to be transformed
+  - The arguments passed to the function are:
+    - The current time as `{ seconds: u64, nanoseconds: u32 }` 
+    - The message `{ topic: string, payload: string }`
+    - The config as read from the pipeline config or updated by the script
+  - The function is expected to return zero, one or many transformed messages `[{ topic: string, payload: string }]`
+  - An exception can be thrown if the input message cannot be transformed.
+- A filter can also export an `update_config` function
+  - This function is called on each message received on the `meta_topics` as defined in the config.
+  - The arguments are:
+    - The message to be interpreted as a config update `{ topic: string, payload: string }`
+    - The current config
+   - The returned value (an arbitrary JSON value) is then used as the new config for the filter.
+- A filter can also export a `tick` function
+  - This function is called at a regular pace with the current time and config.
+  - The filter can then return zero, one or many transformed messages
+  - By sharing an internal state between the `process` and `tick` functions,
+    the filter can implement aggregations over a time window.
+    When messages are received they are pushed by the `process` function into that state
+    and the final outcome is extracted by the `tick` function at the end of the time window.
+
+## First release
+
+While the POC provides a generic mapper that is fully independent of the legacy mappers,
+the plan is not to abandon the latter in favor of the former
+but to revisit the legacy mappers to include the ability for users to add their own mapping rules.
+
+To be lovable, the first release of an extensible mapper should at least:
+
+- be a drop-in replacement of the current mapper (for c8y, aws, az or collect)
+- feature the ability to customize MEA processing by combining builtin filters with user-provided functions written in JavaScript
+- provide tools to create, test, monitor and debug filters and pipelines
+- be stable enough that user-defined filters will still work without changes with future releases.
+
+To keep things simple for the first release, the following questions are deferred:
+
+- Could a generic mapper let users define bridge rules as well as message transformation pipelines?
+- Does it make sense to run such a mapper on child-devices?
+- Could a pipeline send HTTP messages? Or could a filter tell the runtime to send messages over HTTP?
+- How to handle binary payloads on the MQTT bus? 
```

Comment:
Whilst it might not be a focus, we should not prevent users from using binary payloads